plugins/askrene/refine.c

```diff
@@ -632,3 +632,31 @@ double flows_probability(const tal_t *ctx, struct route_query *rq,
 	tal_free(working_ctx);
 	return probability;
 }
+
+bool remove_flows(const tal_t *ctx, struct flow ***flows, u32 n)
+{
+	const tal_t *working_ctx = tal(ctx, tal_t);
+	size_t *flows_index;
+
+	if (n == 0)
+		goto fail;
+	if (n > tal_count(*flows))
+		goto fail;
+
+	flows_index = tal_arrz(working_ctx, size_t, tal_count(*flows));
+	for (size_t i = 0; i < tal_count(*flows); i++) {
+		flows_index[i] = i;
+	}
+	asort(flows_index, tal_count(flows_index), revcmp_flows, *flows);
+	for (size_t count = tal_count(flows_index); n > 0; n--, count--) {
+		assert(count > 0);
+		tal_arr_remove(&flows_index, count - 1);
+	}
+	write_selected_flows(working_ctx, flows_index, flows);
```

Comment:
The use of indexes here (and elsewhere) is weird.  You can just sort the flows directly: they are pointers. 

eg.

```
size_t remaining = tal_count(*flows) - n;

assert(n <= tal_count(*flows));

/* Sort largest to smallest */
asort(flows, tal_count(flows), revcmp_flow, NULL);
for (size_t i = remaining; i < tal_count(*flows); i++)
    tal_free((*flows)[i]);
tal_resize(flows, remaining);
```

The other uses of write_selected_flows are equally suspicious!

Reply:
Fixed.